src/adapter/src/catalog.rs

```diff
@@ -2714,7 +2714,16 @@ impl Catalog {
         // table updates and most-recent-timestamp calculations on a single
         // iterator.
         let storage_usage_events = catalog.storage().await.storage_usage().await?;
-        for event in storage_usage_events {
+        // If no usage retention period is set, we set it to an unreasonably large number
+        // of milliseconds, so that the filter code is simpler.
+        // TODO(necaris): remove this `as`. 584_542_046 years should be enough retention for anyone
+        let retention_period_ms = config
+            .storage_usage_retention_period
+            .map(|p| p.as_millis() as u64)
```

Comment:
Keep this as a u128 which removes the `as` (we have banned `as`). Then use `u128::from(e.timestamp())` in the comparison below.